### PR TITLE
Blog post for Dec 2022 refresh of Liberty Tools

### PR DESCRIPTION
**Pull request by yeekangc**
Blog post for early release refresh of Liberty Tools.  This is for #2796.

### CODE REVIEW

**Comment by yeekangc** [comment]
Targeting 2022/12/19 for publishing for now.  We can align with the site refresh for 22.0.0.13.  The earliest we can go is 2022/12/16.

**Comment by yeekangc** [comment]
Cc @natalie-bernhard @kinueng

File: posts/2022-12-19-cloud-native-java-development-with-liberty-tools.adoc [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+// image::img/blog/FILE_NAME[IMAGE CAPTION ,width=70%,align="center"]
+// -------------------------
+// "[.img_border_light]" = This adds a faint grey border around the image to make its edges sharper. Use it around screenshots but not           
+// around diagrams. Then double check how it looks.
+// There is also a "[.img_border_dark]" class which tends to work best with screenshots that are taken on dark backgrounds.
+// Change "FILE_NAME" to the name of the image file. Also make sure to put the image into the right folder which is: img/blog
+// change the "IMAGE CAPTION" to a couple words of what the image is
+// // // // // // // //
+
+Developers want an intuitive set of tools that allows them to focus on their code, simplifies their iterative coding cycle, supports the APIs they use, takes away mundane manual tasks, gives them flexibility in IDE choices, and yet isn’t in their way but there when they need it.  Essentially, a delightful set of developer tools for cloud-native application development. Enter, the Liberty Tools, as extensions to the popular Java IDEs: Eclipse IDE, Visual Studio Code, and IntelliJ IDEA.
```
Comment (dmuelle) [inline posts/2022-12-19-cloud-native-java-development-with-liberty-tools.adoc]:
```suggestion
Developers need tools  for cloud-native application development that are both intuitive and delightful. Enter Liberty Tools: a set developer tools for the Eclipse IDE, Visual Studio Code, and IntelliJ IDEA development environments.

Liberty Tools can simplify your iterative coding cycle by taking away mundane manual tasks and providing coding support for the APIs you use. They also give you flexibility in your choice of development environment.
```

Comment (dmuelle) [reply posts/2022-12-19-cloud-native-java-development-with-liberty-tools.adoc]: IMO this sentence is trying to do too much- breaking it up improves readability

File: posts/2022-12-19-cloud-native-java-development-with-liberty-tools.adoc [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+Liberty Tools bring together several capabilities for developers:
```
Comment (dmuelle) [inline posts/2022-12-19-cloud-native-java-development-with-liberty-tools.adoc]:
```suggestion
Liberty Tools bring together a range of capabilities:
```

File: posts/2022-12-19-cloud-native-java-development-with-liberty-tools.adoc [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+* Iterate fast using Liberty dev mode integrated within the IDE: Simply start dev mode then write your code. With dev mode, we do the heavy lifting for you: picking up your code changes, running them so that you can test and debug (then repeat)
```
Comment (dmuelle) [inline posts/2022-12-19-cloud-native-java-development-with-liberty-tools.adoc]:
```suggestion
* Iterate fast with Liberty dev mode integrated within your IDE. With dev mode, we do the heavy lifting for you: code changes are automatically detected, recompiled, and deployed whenever you save a new change.
```

File: posts/2022-12-19-cloud-native-java-development-with-liberty-tools.adoc [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+* Write code confidently using Jakarta EE and Jakarta EE APIs with assistance (e.g., validations, quick fixes, completions)
```
Comment (dmuelle) [inline posts/2022-12-19-cloud-native-java-development-with-liberty-tools.adoc]:
```suggestion
* Write code confidently using Jakarta EE and Jakarta EE APIs with assistance such as validations, quick fixes, completions.
```

File: posts/2022-12-19-cloud-native-java-development-with-liberty-tools.adoc [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+* Edit Liberty configuration files easily with assistance
```
Comment (dmuelle) [inline posts/2022-12-19-cloud-native-java-development-with-liberty-tools.adoc]:
```suggestion
* Edit Liberty configuration files easily with assistance.
```

File: posts/2022-12-19-cloud-native-java-development-with-liberty-tools.adoc [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+* Build applications to run on Liberty with Maven and Gradle using Liberty Maven and Gradle plugins
```
Comment (dmuelle) [inline posts/2022-12-19-cloud-native-java-development-with-liberty-tools.adoc]:
```suggestion
* Use the Liberty Maven and Gradle plug-ins to build Liberty applications.
```

File: posts/2022-12-19-cloud-native-java-development-with-liberty-tools.adoc [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+
+Today, we are delighted to share that we are making an early release refresh of the Liberty Tools with a core set of capabilities for building cloud-native applications with Liberty available to Java developers supporting Jakarta EE 9.x Web Profile and MicroProfile 3 & 4 APIs.  Try them out:
```
Comment (dmuelle) [inline posts/2022-12-19-cloud-native-java-development-with-liberty-tools.adoc]:
```suggestion
Today, we are delighted to share that an early release refresh of the Liberty Tools. This release provides a core set of capabilities for building cloud-native applications with Liberty, including support for Jakarta EE 9.x Web Profile and MicroProfile 3 & 4 APIs.  Try them out:
```

File: posts/2022-12-19-cloud-native-java-development-with-liberty-tools.adoc [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+
+* https://marketplace.eclipse.org/content/liberty-tools[Download] Liberty Tools for Eclipse IDE
```
Comment (dmuelle) [inline posts/2022-12-19-cloud-native-java-development-with-liberty-tools.adoc]:
```suggestion
* https://marketplace.eclipse.org/content/liberty-tools[Liberty Tools for Eclipse IDE] 
```

File: posts/2022-12-19-cloud-native-java-development-with-liberty-tools.adoc [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+* https://marketplace.visualstudio.com/items?itemName=Open-Liberty.liberty-dev-vscode-ext[Download] Liberty Tools for Visual Studio Code
```
Comment (dmuelle) [inline posts/2022-12-19-cloud-native-java-development-with-liberty-tools.adoc]:
```suggestion
* https://marketplace.visualstudio.com/items?itemName=Open-Liberty.liberty-dev-vscode-ext[Liberty Tools for Visual Studio Code] 
```

File: posts/2022-12-19-cloud-native-java-development-with-liberty-tools.adoc [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+* https://plugins.jetbrains.com/plugin/14856-liberty-tools[Download] Liberty Tools for IntelliJ IDEA
```
Comment (dmuelle) [inline posts/2022-12-19-cloud-native-java-development-with-liberty-tools.adoc]:
```suggestion
* https://plugins.jetbrains.com/plugin/14856-liberty-tools[Liberty Tools for IntelliJ IDEA] 
```

File: posts/2022-12-19-cloud-native-java-development-with-liberty-tools.adoc [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+
+Do let us know what you think.  Your feedback is important as we round out the capabilities and graduate Liberty Tools from early release status!
+
+* https://github.com/OpenLiberty/liberty-tools-eclipse/issues[Open issues] for Liberty Tools for Eclipse IDE
+* https://github.com/OpenLiberty/liberty-tools-vscode/issues[Open issues] for Liberty Tools for Visual Studio Code
+* https://github.com/OpenLiberty/liberty-tools-intellij/issues[Open issues] for Liberty Tools for IntelliJ IDEA
+
+We will dive into the tools further in future blog posts.  Please stay tuned.
```
Comment (dmuelle) [inline posts/2022-12-19-cloud-native-java-development-with-liberty-tools.adoc]:
```suggestion
We will dive into the tools further in future blog posts.  Stay tuned!
```

**Review by dmuelle — COMMENTED** [review]
ID review- a few suggestions

File: posts/2022-12-19-cloud-native-java-development-with-liberty-tools.adoc [hunk 12]
```diff
@@ -0,0 +1,84 @@
...
+categories: blog
+author_picture: https://avatars3.githubusercontent.com/yeekangc
+author_github: https://github.com/yeekangc
+seo-title: Fast iterative cloud-native Java application development using Liberty Tools - OpenLiberty.io
+seo-description: Liberty Tools enable fast iterative development of cloud-native applications using Jakarta EE & MicroProfile APIs in Eclipse IDE, IntelliJ IDEA, and Visual Studio Code for Java developers
+blog_description: "Liberty Tools enable fast iterative development of cloud-native applications using Jakarta EE & MicroProfile APIs in Eclipse IDE, IntelliJ IDEA, and Visual Studio Code for Java developers"
+open-graph-image: https://openliberty.io/img/twitter_card.jpg
+open-graph-image-alt: Open Liberty Logo
+---
+= Fast iterative cloud-native Java application development using Liberty Tools
```
Comment (dmuelle) [inline posts/2022-12-19-cloud-native-java-development-with-liberty-tools.adoc]:
```suggestion
= Fast iterative cloud-native Java application development with Liberty Tools
```

File: posts/2022-12-19-cloud-native-java-development-with-liberty-tools.adoc [hunk 13]
```diff
@@ -0,0 +1,84 @@
+---
+layout: post
+title: "Fast iterative cloud-native Java application development using Liberty Tools"
+# Do NOT change the categories section
+categories: blog
+author_picture: https://avatars3.githubusercontent.com/yeekangc
+author_github: https://github.com/yeekangc
+seo-title: Fast iterative cloud-native Java application development using Liberty Tools - OpenLiberty.io
```
Comment (dmuelle) [inline posts/2022-12-19-cloud-native-java-development-with-liberty-tools.adoc]:
```suggestion
seo-title: Fast iterative cloud-native Java application development with Liberty Tools - OpenLiberty.io
```

File: posts/2022-12-19-cloud-native-java-development-with-liberty-tools.adoc [hunk 14]
```diff
@@ -0,0 +1,84 @@
...
+seo-description: Liberty Tools enable fast iterative development of cloud-native applications using Jakarta EE & MicroProfile APIs in Eclipse IDE, IntelliJ IDEA, and Visual Studio Code for Java developers
```
Comment (dmuelle) [inline posts/2022-12-19-cloud-native-java-development-with-liberty-tools.adoc]:
```suggestion
seo-description: Liberty Tools enable fast iterative development of cloud-native applications that use Jakarta EE & MicroProfile APIs in Eclipse IDE, IntelliJ IDEA, and Visual Studio Code for Java developers
```

File: posts/2022-12-19-cloud-native-java-development-with-liberty-tools.adoc [hunk 15]
```diff
@@ -0,0 +1,84 @@
...
+blog_description: "Liberty Tools enable fast iterative development of cloud-native applications using Jakarta EE & MicroProfile APIs in Eclipse IDE, IntelliJ IDEA, and Visual Studio Code for Java developers"
```
Comment (dmuelle) [inline posts/2022-12-19-cloud-native-java-development-with-liberty-tools.adoc]:
```suggestion
blog_description: "Liberty Tools enable fast iterative development of cloud-native applications that use Jakarta EE & MicroProfile APIs in Eclipse IDE, IntelliJ IDEA, and Visual Studio Code for Java developers"
```

File: posts/2022-12-19-cloud-native-java-development-with-liberty-tools.adoc [hunk 16]
```diff
@@ -0,0 +1,84 @@
+---
+layout: post
+title: "Fast iterative cloud-native Java application development using Liberty Tools"
```
Comment (dmuelle) [inline posts/2022-12-19-cloud-native-java-development-with-liberty-tools.adoc]:
```suggestion
title: "Fast iterative cloud-native Java application development with Liberty Tools"
```

**Review by dmuelle — COMMENTED** [review]
a few more comments

**Comment by yeekangc** [comment]
Thank you for the review, @dmuelle.  I committed your suggestions and made a few updates.  Can you please take another look?

File: posts/2022-12-19-cloud-native-java-development-with-liberty-tools.adoc [hunk 17]
```diff
@@ -0,0 +1,86 @@
...
+// // // // // // // //
+
+Developers need tools for cloud-native Java application development that are both intuitive and delightful. Enter Liberty Tools: the next-generation tools for developers to build applications with Liberty within the Eclipse IDE, Visual Studio Code, and IntelliJ IDEA integrated development environments (IDE).
+
+Liberty Tools can simplify your iterative coding cycle by taking away mundane manual tasks and providing coding support for the APIs you use. They also give you flexibility in your choice of IDE.
+
+Liberty Tools bring together a range of capabilities:
+
+* Iterate fast with Liberty dev mode integrated within your IDE. With dev mode, we do the heavy lifting for you: code changes are automatically detected, recompiled, and deployed whenever you save a new change. You can test and debug right away.
+* Write code confidently using Jakarta EE and Jakarta EE APIs with assistance such as validations, quick fixes, completions.
```
Comment (dmuelle) [inline posts/2022-12-19-cloud-native-java-development-with-liberty-tools.adoc]:
```suggestion
* Write code confidently using MicroProfile and Jakarta EE APIs with assistance such as validations, quick fixes, completions.
```

**Review by dmuelle — CHANGES_REQUESTED** [review]
replace extra mention of Jakarta with MP

**Review by dmuelle — DISMISSED** [review]
LGTM- if we are targeting Tuesday to publish, need to change the file name to 
`2022-12-20-cloud-native-java-development-with-liberty-tools.adoc`

**Comment by dmuelle** [comment]
@yeekangc - @mbroz2 suggested on slack we could add these images of the tools offerings in the 3 marketplaces to make the post more appealing

lt-ecl
lt-intj
lt-vsc

If you don't have a chance to incorporate them when you update the filename, I can add them before we publish. Just don't want to create a conflict in versions since I dont have access to your fork

**Review by dmuelle — APPROVED** [review]
LGTM

**Pull request merged**